teachertool/src/global.scss

```diff
@@ -0,0 +1,347 @@
+/*******************************/
+/*****  GLOBAL VARIABLES  ******/
+/*******************************/
+
+:root {
+    --frame-loader-zindex: 30;
+    --above-frame-zindex: 50;
+    --modal-dimmer-zindex: 100;
+    --fullscreen-modal-zindex: 90;
+}
+
+/*******************************/
+/*********  RESETS  ************/
+/*******************************/
+
+* {
+    // undo semantic.css defaults
+    min-height: initial;
+    min-width: initial;
+}
+
+*,
+*::before,
+*::after {
+    box-sizing: border-box;
+}
+
+* {
+    margin: 0;
+}
+
+img,
+picture,
+video,
+canvas,
+svg {
+    display: block;
+    max-width: 100%;
+}
+
+input,
+button,
+textarea,
+select {
+    font: inherit;
+}
+
+p,
+h1,
+h2,
+h3,
+h4,
+h5,
+h6 {
+    overflow-wrap: break-word;
+}
+
+body {
+    -webkit-font-smoothing: antialiased;
+    -moz-osx-font-smoothing: grayscale;
+    font-family: var(--pxt-page-font);
+    background: var(--pxt-page-background);
+    color: var(--pxt-page-foreground);
+    line-height: 1.5;
+    margin: 0;
+    padding: 0;
+    overflow: hidden;
+    // undo semantic.css defaults
+    display: initial;
+    font-size: initial;
+}
+
+code {
+    font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New", monospace;
+}
+
+/*******************************/
+/*********  GLOBALS  ***********/
+/*******************************/
+
+#root {
+    display: flex;
+    flex-direction: column;
+    height: 100%;
+    width: 100%;
+    overflow: hidden;
+    isolation: isolate;
+}
+
+.noclick {
+    pointer-events: none;
+    cursor: default;
+}
+
+/**********************************************/
+/********* THINGS THAT NEED HOMES *************/
+/**********************************************/
+
+.btn-primary {
```

Comment:
So is the idea that ultimately these kinds of styles (referencing the theme variables) will move into react common, so this doesn't have to be repeated in every app's global css?

Reply:
Yes, it means that classes like "primary" can move to react common.

Reply:
Isn't primary already in react common?

Reply:
If so it will need to be updated use the color vars, so it can be changed at runtime.